### DISCUSSION TITLE
Ignore Specific Field(s) on POJO in all mapping implementations Mapstruct

### DISCUSSION THREAD

**Discussion by shameekagarwal** [mapstruct/mapstruct#3147] [Q&A] [answered]
My persistent entities has fields `createdBy`, `createdAt`, `modifiedBy`, `modifiedAt`.    
I would like mapstruct to **always** ignore mapping these fields, since they are autogenerated using hibernate annotations.  
I don't like the idea of doing `@Mapping(target = ..., ignore = true)` repeatedly.

Is there a way to do this at the, for e.g. at my entity level, instead of the Mapper method level?  
Or is there some other better solution than repeating the annotations?

bit more context - the mapping implementations use builders generated by lombok.  
my entites can either extend `UpdatableEntity` or `BaseEntity`  
all my entities, `UpdatableEntity` and `BaseEntity` use `@SuperBuilder`  
`BaseEntity` has `createdBy` and `createdAt`  
`UpdatableEntity` has `modifiedBy` and `modifiedAt` and extends `BaseEntity`

**Comment by filiphr** [comment, marked as answer]
Currently there are 2 possible ways to achieve what you need.

## Mapping Composition

MapStruct supports mapping composition for `@Mapping`. This means that you can define your own annotation `@MyDefaultIgnores` and would be annotated with `@Mapping(target = ..., ignore= true)` and then you add that annotation to your mapping methods.

## Custom `AccessorNamingStrategy`

MapStruct has an SPI that allows to provide a [Custom Accessor Naming Strategy](https://mapstruct.org/documentation/stable/reference/html/#_custom_accessor_naming_strategy) that allows you to define your own logic about what is considered a getter / setter for MapStruct. You can use that to ignore specific properties from your domain.
